west: Update hal_stm32 module to head following cube updates

erwango opened the pull request:
This update contains the following changes

* Introduce STM32G0 Cube V1.2.0
* Update API for STM32MP1 IPCC
* Update Cube version for following series:
STM32F0XX: from version: V1.9.0 to version: V1.10.1
STM32F1XX: from version: V1.6.1 to version: V1.7.0
STM32F4XX: from version: V1.21.0 to version: V1.24.1
STM32F7XX: from version: V1.12.0 to version: V1.15.1
STM32L0XX: from version: V1.10.0 to version: V1.11.2
STM32L1XX: from version: V1.8.0 to version: V1.9.0
STM32L4XX: from version: V1.13.0 to version: V1.14.0
STM32WBXX: from version: V1.0.0 to version: V1.1.1


Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango commented:
This PR contains update to head of hal_stm32 modules.
It replaces following commits:
https://github.com/zephyrproject-rtos/zephyr/pull/16983
https://github.com/zephyrproject-rtos/zephyr/pull/16976
https://github.com/zephyrproject-rtos/zephyr/pull/16975
https://github.com/zephyrproject-rtos/zephyr/pull/16974
https://github.com/zephyrproject-rtos/zephyr/pull/16960
https://github.com/zephyrproject-rtos/zephyr/pull/16959
https://github.com/zephyrproject-rtos/zephyr/pull/16956
https://github.com/zephyrproject-rtos/zephyr/pull/16955
Wait https://github.com/zephyrproject-rtos/zephyr/pull/16983 CI to complete before Merge

It also includes dependency for https://github.com/zephyrproject-rtos/zephyr/pull/16962 and https://github.com/zephyrproject-rtos/zephyr/pull/16873.

erwango commented:
As this change imapct all STM32 based targets, CI will likely fail.
All series have beenpushed independantly and got CI passed (except #16962 which is a different case).
Remaining one is STM32L4 (#16983), so when checks are passed on it, it is safe to merge current one.

nashif commented:
> As this change imapct all STM32 based targets, CI will likely fail.

it should not fail.

erwango commented:
> it should not fail.

I meant time out, sorry.
Shouldn't it trigger full non regression on all boards impacted by the change, so in this case all STM32 based boards? This is why I'd expect it to time out.
Unless it does not build against all impacted boards for some reason I don't get.

nashif commented:
> Unless it does not build against all impacted boards for some reason I don't get.

this happens only if you actually change the board files directly.

erwango commented:
> this happens only if you actually change the board files directly.

ouch!

erwango commented:
Waits https://github.com/zephyrproject-rtos/hal_stm32/pull/13 to be merged.

erwango committed:
west: Update hal_stm32 module to head following cube updates

This update contains the following changes

* Introduce STM32G0 Cube V1.2.0
* Update API for STM32MP1 IPCC
* Update Cube version for following series:
STM32F0XX: from version: V1.9.0 to version: V1.10.1
STM32F1XX: from version: V1.6.1 to version: V1.7.0
STM32F4XX: from version: V1.21.0 to version: V1.24.1
STM32F7XX: from version: V1.12.0 to version: V1.15.1
STM32L0XX: from version: V1.10.0 to version: V1.11.2
STM32L1XX: from version: V1.8.0 to version: V1.9.0
STM32L4XX: from version: V1.13.0 to version: V1.14.0
STM32WBXX: from version: V1.0.0 to version: V1.1.1
* Includes a fix for L4 following package update
* Includes a fix for G0 following package introduction

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>
